[Automation] java run fails for io.smallrye.common:smallrye-common-process:2.17.0

## java run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.17.0" ./gradlew clean javaTest -Pcoordinates="io.smallrye.common:smallrye-common-process:2.16.0"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25260953949/job/74068130884
Last 300 lines of the log:
```console
io_smallrye_common.smallrye_common_process.Smallrye_common_processTest > outputCanBeCopiedTransferredAndProcessed() SUCCESSFUL

io_smallrye_common.smallrye_common_process.Smallrye_common_processTest > errorStreamCanBeRedirectedOrCapturedOnFailure() SUCCESSFUL

io_smallrye_common.smallrye_common_process.Smallrye_common_processTest > execToStringAndStringListCaptureStandardOutput() SUCCESSFUL

io_smallrye_common.smallrye_common_process.Smallrye_common_processTest > processUtilityCanObserveAndDestroyPlainJdkProcesses() SUCCESSFUL

io_smallrye_common.smallrye_common_process.Smallrye_common_processTest > errorLinesCanBeConsumedWithoutChangingOutputResult() SUCCESSFUL

io_smallrye_common.smallrye_common_process.Smallrye_common_processTest > exitCodeCheckerCanAcceptNonZeroExitStatus() SUCCESSFUL

io_smallrye_common.smallrye_common_process.Smallrye_common_processTest > asynchronousExecutionInvokesWhileRunningHandler() FAILED


Failures (1):
  JUnit Jupiter:Smallrye_common_processTest:asynchronousExecutionInvokesWhileRunningHandler()
    MethodSource [className = 'io_smallrye_common.smallrye_common_process.Smallrye_common_processTest', methodName = 'asynchronousExecutionInvokesWhileRunningHandler', methodParameterTypes = '']
    => java.util.concurrent.TimeoutException
       java.base@25.0.3/java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1981)
       java.base@25.0.3/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2116)
       io_smallrye_common.smallrye_common_process.Smallrye_common_processTest.asynchronousExecutionInvokesWhileRunningHandler(Smallrye_common_processTest.java:306)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)

       org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
FAILURE: Build failed with an exception.
       java.base@25.0.3/java.util.concurrent.FutureTask.run(FutureTask.java:328)

       java.base@25.0.3/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
       [...]

Test run finished after 10053 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[        14 tests found           ]
[         0 tests skipped         ]
[        14 tests started         ]
[         0 tests aborted         ]
[        13 tests successful      ]
[         1 tests failed          ]


> Task :nativeTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.smallrye.common/smallrye-common-process/2.16.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
17 actionable tasks: 9 executed, 8 up-to-date
* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.smallrye.common/smallrye-common-process/2.16.0/build/native/nativeTestCompile/io.smallrye.common.smallrye-common-process_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 40s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for io.smallrye.common:smallrye-common-process:2.16.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 48s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 2.17.0 failed with exit code 1.
SOFT-FAIL[native-image run][2.17.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.17.0" ./gradlew clean test -Pcoordinates="io.smallrye.common:smallrye-common-process:2.16.0"]
========================================================================================
 io.smallrye.common:smallrye-common-process:2.16.0:2.17.0 stage javac compile
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.smallrye.common/smallrye-common-process/2.16.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 866ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileTestJava

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.smallrye.common/smallrye-common-process/2.16.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1s
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 3s
11 actionable tasks: 3 executed, 8 up-to-date
========================================================================================
 io.smallrye.common:smallrye-common-process:2.16.0:2.17.0 stage java run
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.smallrye.common/smallrye-common-process/2.16.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 822ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :compileTestJava
> Task :processTestResources NO-SOURCE
> Task :testClasses

> Task :test FAILED


14 tests completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.smallrye.common/smallrye-common-process/2.16.0/build/reports/tests/test/index.html

* Try:
> Run with --scanSmallrye_common_processTest > asynchronousExecutionInvokesWhileRunningHandler() FAILED
    java.util.concurrent.TimeoutException
        at java.base/java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1960)
        at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2095)
        at io_smallrye_common.smallrye_common_process.Smallrye_common_processTest.asynchronousExecutionInvokesWhileRunningHandler(Smallrye_common_processTest.java:306)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.smallrye.common/smallrye-common-process/2.16.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 3 executed, 10 up-to-date
 to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 12s

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 14s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'java run' for 2.17.0 failed with exit code 1.
FAILED[java run][2.17.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.17.0" ./gradlew clean javaTest -Pcoordinates="io.smallrye.common:smallrye-common-process:2.16.0"]
```

--- a/metadata/io.smallrye.common/smallrye-common-process/2.17.0/reachability-metadata.json
+++ b/metadata/io.smallrye.common/smallrye-common-process/2.17.0/reachability-metadata.json
@@ -1,0 +1,68 @@
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "io.smallrye.common.process.Logging"
+      },
+      "type": "ch.qos.logback.classic.Logger"
+    },
+    {
+      "condition": {
+        "typeReached": "io.smallrye.common.process.Logging"
+      },
+      "type": "io.smallrye.common.process.Logging_$logger",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": [
+            "org.jboss.logging.Logger"
+          ]
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "io.smallrye.common.process.Logging"
+      },
+      "type": "io.smallrye.common.process.Logging_$logger_en"
+    },
+    {
+      "condition": {
+        "typeReached": "io.smallrye.common.process.Logging"
+      },
+      "type": "io.smallrye.common.process.Logging_$logger_en_US"
+    },
+    {
+      "condition": {
+        "typeReached": "io.smallrye.common.process.Logging"
+      },
+      "type": "org.apache.log4j.LogManager"
+    },
+    {
+      "condition": {
+        "typeReached": "io.smallrye.common.process.Logging"
+      },
+      "type": "org.apache.logging.log4j.Logger"
+    },
+    {
+      "condition": {
+        "typeReached": "io.smallrye.common.process.Logging"
+      },
+      "type": "org.jboss.logmanager.LogManager"
+    }
+  ],
+  "resources": [
+    {
+      "condition": {
+        "typeReached": "io.smallrye.common.process.Logging_$logger"
+      },
+      "glob": "META-INF/services/java.time.zone.ZoneRulesProvider"
+    },
+    {
+      "condition": {
+        "typeReached": "io.smallrye.common.process.Logging"
+      },
+      "glob": "META-INF/services/org.jboss.logging.LoggerProvider"
+    }
+  ]
+}

--- a/metadata/io.smallrye.common/smallrye-common-process/index.json
+++ b/metadata/io.smallrye.common/smallrye-common-process/index.json
@@ -2,6 +2,11 @@
   {
     "latest" : true,
     "metadata-version" : "2.17.0",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/io/smallrye/common/smallrye-common-process/$version$/smallrye-common-process-$version$-sources.jar",
+    "repository-url" : "https://github.com/smallrye/smallrye-common",
+    "test-code-url" : "https://github.com/smallrye/smallrye-common/tree/$version$/process/src/test/java",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/io/smallrye/common/smallrye-common-process/$version$/smallrye-common-process-$version$-javadoc.jar",
+    "description" : "SmallRye Common Process provides Java utilities for safely launching subprocesses and managing their I/O. It also exposes helpers for inspecting the current Java process, reducing process-control and stream-handling boilerplate for JVM applications.",
     "tested-versions" : [
       "2.17.0"
     ],

--- a/metadata/io.smallrye.common/smallrye-common-process/index.json
+++ b/metadata/io.smallrye.common/smallrye-common-process/index.json
@@ -1,6 +1,15 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "2.17.0",
+    "tested-versions" : [
+      "2.17.0"
+    ],
+    "allowed-packages" : [
+      "io.smallrye.common.process"
+    ]
+  },
+  {
     "metadata-version" : "2.16.0",
     "source-code-url" : "https://repo.maven.apache.org/maven2/io/smallrye/common/smallrye-common-process/$version$/smallrye-common-process-$version$-sources.jar",
     "repository-url" : "https://github.com/smallrye/smallrye-common",

--- a/stats/io.smallrye.common/smallrye-common-process/2.17.0/execution-metrics.json
+++ b/stats/io.smallrye.common/smallrye-common-process/2.17.0/execution-metrics.json
@@ -1,0 +1,87 @@
+{
+  "fix_java_run_fail:2026-05-05": {
+    "timestamp": "2026-05-05T08:59:13.396708Z",
+    "library": "io.smallrye.common:smallrye-common-process:2.17.0",
+    "starting_commit": "8fd2b341eb32e540530a22baf9d1f2d0bfc8e225",
+    "ending_commit": "6b25b188b5412ab6b206a8baa6aedc952d114aab",
+    "previous_library": "io.smallrye.common:smallrye-common-process:2.16.0",
+    "strategy_name": "java_run_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "2.17.0",
+      "dynamicAccess": {
+        "breakdown": {},
+        "coverageRatio": 1.0,
+        "coveredCalls": 0,
+        "totalCalls": 0
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 3773,
+          "missed": 2306,
+          "ratio": 0.620661,
+          "total": 6079
+        },
+        "line": {
+          "covered": 898,
+          "missed": 541,
+          "ratio": 0.624044,
+          "total": 1439
+        },
+        "method": {
+          "covered": 206,
+          "missed": 139,
+          "ratio": 0.597101,
+          "total": 345
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "2.16.0",
+      "dynamicAccess": {
+        "breakdown": {},
+        "coverageRatio": 1.0,
+        "coveredCalls": 0,
+        "totalCalls": 0
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 3773,
+          "missed": 2306,
+          "ratio": 0.620661,
+          "total": 6079
+        },
+        "line": {
+          "covered": 898,
+          "missed": 541,
+          "ratio": 0.624044,
+          "total": 1439
+        },
+        "method": {
+          "covered": 206,
+          "missed": 139,
+          "ratio": 0.597101,
+          "total": 345
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 73761,
+      "cached_input_tokens_used": 1357312,
+      "output_tokens_used": 7854,
+      "iterations": 1,
+      "cost_usd": 0.9143,
+      "tested_library_loc": 898,
+      "metadata_entries": 10,
+      "previous_library_metadata_entries": 10,
+      "code_coverage_percent": 62.4,
+      "previous_library_coverage_percent": 62.4
+    },
+    "artifacts": {
+      "test_file": "tests/src/io.smallrye.common/smallrye-common-process/2.17.0/src/test/java/io_smallrye_common/smallrye_common_process/Smallrye_common_processTest.java",
+      "metadata_file": "metadata/io.smallrye.common/smallrye-common-process/2.17.0/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/io.smallrye.common/smallrye-common-process/2.17.0/stats.json
+++ b/stats/io.smallrye.common/smallrye-common-process/2.17.0/stats.json
@@ -1,0 +1,31 @@
+{
+  "versions" : [ {
+    "version" : "2.17.0",
+    "dynamicAccess" : {
+      "breakdown" : { },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 0,
+      "totalCalls" : 0
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 3773,
+        "missed" : 2306,
+        "ratio" : 0.620661,
+        "total" : 6079
+      },
+      "line" : {
+        "covered" : 898,
+        "missed" : 541,
+        "ratio" : 0.624044,
+        "total" : 1439
+      },
+      "method" : {
+        "covered" : 206,
+        "missed" : 139,
+        "ratio" : 0.597101,
+        "total" : 345
+      }
+    }
+  } ]
+}

--- a/tests/src/io.smallrye.common/smallrye-common-process/2.17.0/.gitignore
+++ b/tests/src/io.smallrye.common/smallrye-common-process/2.17.0/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/io.smallrye.common/smallrye-common-process/2.17.0/build.gradle
+++ b/tests/src/io.smallrye.common/smallrye-common-process/2.17.0/build.gradle
@@ -1,0 +1,27 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "io.smallrye.common:smallrye-common-process:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/io.smallrye.common/smallrye-common-process/2.17.0/gradle.properties
+++ b/tests/src/io.smallrye.common/smallrye-common-process/2.17.0/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = io.smallrye.common:smallrye-common-process:2.17.0
+library.version = 2.17.0
+metadata.dir = io.smallrye.common/smallrye-common-process/2.17.0/

--- a/tests/src/io.smallrye.common/smallrye-common-process/2.17.0/settings.gradle
+++ b/tests/src/io.smallrye.common/smallrye-common-process/2.17.0/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'io.smallrye.common.smallrye-common-process_tests'

--- a/tests/src/io.smallrye.common/smallrye-common-process/2.17.0/src/test/java/io_smallrye_common/smallrye_common_process/Smallrye_common_processTest.java
+++ b/tests/src/io.smallrye.common/smallrye-common-process/2.17.0/src/test/java/io_smallrye_common/smallrye_common_process/Smallrye_common_processTest.java
@@ -1,0 +1,345 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package io_smallrye_common.smallrye_common_process;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import io.smallrye.common.process.AbnormalExitException;
+import io.smallrye.common.process.PipelineExecutionException;
+import io.smallrye.common.process.ProcessBuilder;
+import io.smallrye.common.process.ProcessUtil;
+import io.smallrye.common.process.WaitableProcessHandle;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class Smallrye_common_processTest {
+    private static final Duration EXIT_TIMEOUT = Duration.ofSeconds(2);
+    private static final Path SHELL = requiredCommand("sh");
+    private static final Path CAT = requiredCommand("cat");
+    private static final Path GREP = requiredCommand("grep");
+
+    @Test
+    void processUtilitiesExposeNativeCharsetJavaCommandAndSearchPath() {
+        assertThat(ProcessUtil.nativeCharset()).isNotNull();
+        assertThat(ProcessUtil.searchPath()).doesNotContainNull();
+        assertThat(ProcessUtil.nameOfJava()).isIn("java", "java.exe");
+
+        Path javaPath = ProcessUtil.pathOfJava();
+        assertThat(javaPath).isNotNull();
+        if (javaPath.isAbsolute()) {
+            assertThat(Files.isExecutable(javaPath)).isTrue();
+        } else {
+            assertThat(javaPath).isEqualTo(Path.of(ProcessUtil.nameOfJava()));
+        }
+
+        assertThat(ProcessUtil.pathOfCommand(SHELL)).contains(SHELL);
+        Path missingCommand = Path.of("smallrye-common-process-command-that-should-not-exist");
+        assertThat(ProcessUtil.pathOfCommand(missingCommand)).isEmpty();
+    }
+
+    @Test
+    void execToStringAndStringListCaptureStandardOutput() {
+        String singleString = ProcessBuilder.execToString(SHELL, shellArguments("printf 'alpha\\nbeta\\n'"));
+        assertThat(singleString).isEqualTo("alpha\nbeta\n");
+
+        List<String> lines = shellCommand("printf 'one\\ntwo\\nthree\\n'")
+                .output()
+                .toStringList(10, 100)
+                .run();
+
+        assertThat(lines).containsExactly("one", "two", "three");
+    }
+
+    @Test
+    void inputFactoriesCanFeedProcessStandardInput() {
+        List<String> fromStrings = ProcessBuilder.newBuilder(CAT)
+                .softExitTimeout(EXIT_TIMEOUT)
+                .hardExitTimeout(EXIT_TIMEOUT)
+                .input()
+                .charset(StandardCharsets.UTF_8)
+                .fromStrings(List.of("first", "second"))
+                .output()
+                .charset(StandardCharsets.UTF_8)
+                .toStringList(5, 100)
+                .run();
+
+        assertThat(fromStrings).containsExactly("first", "second");
+
+        String fromReader = ProcessBuilder.newBuilder(CAT)
+                .softExitTimeout(EXIT_TIMEOUT)
+                .hardExitTimeout(EXIT_TIMEOUT)
+                .input()
+                .charset(StandardCharsets.UTF_8)
+                .transferFrom(new StringReader("reader-input"))
+                .output()
+                .charset(StandardCharsets.UTF_8)
+                .toSingleString(100)
+                .run();
+
+        assertThat(fromReader).isEqualTo("reader-input");
+
+        String fromBytes = ProcessBuilder.newBuilder(CAT)
+                .softExitTimeout(EXIT_TIMEOUT)
+                .hardExitTimeout(EXIT_TIMEOUT)
+                .input()
+                .transferFrom(new ByteArrayInputStream("byte-input".getBytes(StandardCharsets.UTF_8)))
+                .output()
+                .charset(StandardCharsets.UTF_8)
+                .toSingleString(100)
+                .run();
+
+        assertThat(fromBytes).isEqualTo("byte-input");
+    }
+
+    @Test
+    void inputAndOutputCanTransferThroughFiles(@TempDir Path tempDir) throws Exception {
+        Path input = tempDir.resolve("input.txt");
+        Path output = tempDir.resolve("output.txt");
+        Files.writeString(input, "from-file\n", StandardCharsets.UTF_8);
+
+        ProcessBuilder.newBuilder(CAT)
+                .softExitTimeout(EXIT_TIMEOUT)
+                .hardExitTimeout(EXIT_TIMEOUT)
+                .input()
+                .transferFrom(input)
+                .output()
+                .transferTo(output)
+                .run();
+
+        shellCommand("printf 'appended\\n'")
+                .output()
+                .appendTo(output)
+                .run();
+
+        assertThat(Files.readString(output, StandardCharsets.UTF_8)).isEqualTo("from-file\nappended\n");
+    }
+
+    @Test
+    void outputCanBeCopiedTransferredAndProcessed() throws Exception {
+        ByteArrayOutputStream copiedBytes = new ByteArrayOutputStream();
+        String output = shellCommand("printf 'copy-me'")
+                .output()
+                .charset(StandardCharsets.UTF_8)
+                .toSingleString(100)
+                .copyAndTransferTo(copiedBytes)
+                .run();
+
+        assertThat(output).isEqualTo("copy-me");
+        assertThat(copiedBytes.toString(StandardCharsets.UTF_8)).isEqualTo("copy-me");
+
+        StringWriter copiedCharacters = new StringWriter();
+        Integer lineCount = shellCommand("printf 'a\\nb\\nc\\n'")
+                .output()
+                .charset(StandardCharsets.UTF_8)
+                .processWith(reader -> (int) reader.lines().count())
+                .copyAndTransferTo(copiedCharacters)
+                .run();
+
+        assertThat(lineCount).isEqualTo(3);
+        assertThat(copiedCharacters.toString()).isEqualTo("a\nb\nc\n");
+
+        Integer byteCount = shellCommand("printf '12345'")
+                .output()
+                .processBytesWith(inputStream -> inputStream.readAllBytes().length)
+                .run();
+
+        assertThat(byteCount).isEqualTo(5);
+    }
+
+    @Test
+    void environmentAndWorkingDirectoryAreApplied(@TempDir Path tempDir) throws Exception {
+        Path marker = tempDir.resolve("marker.txt");
+        Files.writeString(marker, "marker", StandardCharsets.UTF_8);
+
+        String output = shellCommand("cat marker.txt; printf ':%s' \"$SMALLRYE_PROCESS_TEST_VALUE\"")
+                .directory(tempDir)
+                .environment(System.getenv())
+                .modifyEnvironment(environment -> environment.put("SMALLRYE_PROCESS_TEST_VALUE", "configured"))
+                .output()
+                .toSingleString(100)
+                .run();
+
+        assertThat(output).isEqualTo("marker:configured");
+    }
+
+    @Test
+    void errorStreamCanBeRedirectedOrCapturedOnFailure() {
+        String redirected = shellCommand("printf 'stdout'; printf ':stderr' >&2")
+                .error()
+                .redirect()
+                .output()
+                .toSingleString(100)
+                .run();
+
+        assertThat(redirected).contains("stdout").contains(":stderr");
+
+        AbnormalExitException exception = assertThrows(AbnormalExitException.class, () -> shellCommand(
+                "printf 'out-head\\nout-tail\\n'; printf 'err-head\\nerr-tail\\n' >&2; exit 7")
+                .output()
+                .gatherOnFail(true)
+                .captureHeadLines(1)
+                .captureTailLines(1)
+                .error()
+                .gatherOnFail(true)
+                .captureHeadLines(1)
+                .captureTailLines(1)
+                .run());
+
+        assertThat(exception.exitCode()).isEqualTo(7);
+        assertThat(exception.output()).containsExactly("out-head", "out-tail");
+        assertThat(exception.errorOutput()).containsExactly("err-head", "err-tail");
+        assertThat(exception.command()).isEqualTo(SHELL);
+        assertThat(exception.arguments()).containsExactlyElementsOf(shellArguments(
+                "printf 'out-head\\nout-tail\\n'; printf 'err-head\\nerr-tail\\n' >&2; exit 7"));
+        assertThat(exception.getMessage()).contains("exit code 7", "out-head", "err-head");
+    }
+
+    @Test
+    void exitCodeCheckerCanAcceptNonZeroExitStatus() {
+        String output = shellCommand("printf 'accepted'; exit 7")
+                .exitCodeChecker(exitCode -> exitCode == 7)
+                .output()
+                .toSingleString(100)
+                .run();
+
+        assertThat(output).isEqualTo("accepted");
+    }
+
+    @Test
+    void programmaticInputProducersCanFeedProcessStandardInput() {
+        List<String> producedCharacters = ProcessBuilder.newBuilder(CAT)
+                .softExitTimeout(EXIT_TIMEOUT)
+                .hardExitTimeout(EXIT_TIMEOUT)
+                .input()
+                .charset(StandardCharsets.UTF_8)
+                .produceWith(writer -> writer.write("writer-line-one\nwriter-line-two\n"))
+                .output()
+                .charset(StandardCharsets.UTF_8)
+                .toStringList(5, 100)
+                .run();
+
+        assertThat(producedCharacters).containsExactly("writer-line-one", "writer-line-two");
+
+        String producedBytes = ProcessBuilder.newBuilder(CAT)
+                .softExitTimeout(EXIT_TIMEOUT)
+                .hardExitTimeout(EXIT_TIMEOUT)
+                .input()
+                .produceBytesWith(outputStream -> outputStream.write(
+                        "bytes-produced".getBytes(StandardCharsets.UTF_8)))
+                .output()
+                .charset(StandardCharsets.UTF_8)
+                .toSingleString(100)
+                .run();
+
+        assertThat(producedBytes).isEqualTo("bytes-produced");
+    }
+
+    @Test
+    void errorLinesCanBeConsumedWithoutChangingOutputResult() {
+        List<String> errorLines = new ArrayList<>();
+
+        String output = shellCommand("printf 'stdout-result'; printf 'warning-one\\nwarning-two\\n' >&2")
+                .output()
+                .toSingleString(100)
+                .error()
+                .consumeLinesWith(100, errorLines::add)
+                .run();
+
+        assertThat(output).isEqualTo("stdout-result");
+        assertThat(errorLines).containsExactly("warning-one", "warning-two");
+    }
+
+    @Test
+    void pipelineFeedsOutputIntoNextProcess() {
+        List<String> output = shellCommand("printf 'alpha\\nbravo\\ncharlie\\n'")
+                .output()
+                .pipeTo(GREP, "bravo")
+                .output()
+                .toStringList(5, 100)
+                .run();
+
+        assertThat(output).containsExactly("bravo");
+    }
+
+    @Test
+    void pipelineFailureReportsProcessExceptions() {
+        PipelineExecutionException exception = assertThrows(PipelineExecutionException.class,
+                () -> shellCommand("exit 3")
+                        .output()
+                        .pipeTo(SHELL, "-c", "cat >/dev/null; exit 4")
+                        .run());
+
+        assertThat(exception.processExecutionExceptions())
+                .hasSize(2)
+                .allSatisfy(processException -> assertThat(processException.command()).isEqualTo(SHELL));
+    }
+
+    @Test
+    void asynchronousExecutionInvokesWhileRunningHandler() throws Exception {
+        AtomicReference<WaitableProcessHandle> seenHandle = new AtomicReference<>();
+        CompletableFuture<String> future = shellCommand("printf 'async-output'")
+                .whileRunning(seenHandle::set)
+                .output()
+                .toSingleString(100)
+                .runAsync();
+
+        assertThat(future.get(10, TimeUnit.SECONDS)).isEqualTo("async-output");
+        assertThat(seenHandle.get()).isNotNull();
+        assertThat(seenHandle.get().pid()).isPositive();
+        assertThat(seenHandle.get().command()).isEqualTo(SHELL);
+        assertThat(seenHandle.get().arguments()).containsExactlyElementsOf(shellArguments("printf 'async-output'"));
+        assertThat(seenHandle.get().waitUninterruptiblyFor(1, TimeUnit.SECONDS)).isTrue();
+        assertThat(seenHandle.get().exitValue()).isZero();
+    }
+
+    @Test
+    void processUtilityCanObserveAndDestroyPlainJdkProcesses() throws Exception {
+        java.lang.Process finished = new java.lang.ProcessBuilder(SHELL.toString(), "-c", "exit 0").start();
+        assertThat(ProcessUtil.stillRunningAfter(finished, Duration.ofSeconds(5))).isFalse();
+
+        java.lang.Process running = new java.lang.ProcessBuilder(SHELL.toString(), "-c", "read ignored").start();
+        try {
+            assertThat(ProcessUtil.stillRunningAfter(running, Duration.ofMillis(10))).isTrue();
+            ProcessUtil.destroyAllForcibly(running);
+            assertThat(ProcessUtil.stillRunningAfter(running, Duration.ofSeconds(5))).isFalse();
+        } finally {
+            ProcessUtil.destroyAllForcibly(running);
+        }
+    }
+
+    private static ProcessBuilder<Void> shellCommand(String script) {
+        return ProcessBuilder.newBuilder(SHELL, shellArguments(script))
+                .softExitTimeout(EXIT_TIMEOUT)
+                .hardExitTimeout(EXIT_TIMEOUT);
+    }
+
+    private static List<String> shellArguments(String script) {
+        return List.of("-c", script);
+    }
+
+    private static Path requiredCommand(String command) {
+        Optional<Path> path = ProcessUtil.pathOfCommand(Path.of(command));
+        return path.orElseThrow(
+                () -> new IllegalStateException("Required command is not available on PATH: " + command));
+    }
+}

--- a/tests/src/io.smallrye.common/smallrye-common-process/2.17.0/src/test/java/io_smallrye_common/smallrye_common_process/Smallrye_common_processTest.java
+++ b/tests/src/io.smallrye.common/smallrye-common-process/2.17.0/src/test/java/io_smallrye_common/smallrye_common_process/Smallrye_common_processTest.java
@@ -22,10 +22,12 @@ import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicReference;
 
 import io.smallrye.common.process.AbnormalExitException;
+import io.smallrye.common.process.Logging_$logger;
 import io.smallrye.common.process.PipelineExecutionException;
 import io.smallrye.common.process.ProcessBuilder;
 import io.smallrye.common.process.ProcessUtil;
 import io.smallrye.common.process.WaitableProcessHandle;
+import org.jboss.logging.Logger;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.io.TempDir;
 
@@ -37,6 +39,13 @@ public class Smallrye_common_processTest {
     private static final Path SHELL = requiredCommand("sh");
     private static final Path CAT = requiredCommand("cat");
     private static final Path GREP = requiredCommand("grep");
+
+    @Test
+    void generatedMessageLoggerIsReachableForProcessThreads() {
+        Logging_$logger logger = new Logging_$logger(Logger.getLogger("smallrye-common-process-test"));
+
+        assertThat(logger).isNotNull();
+    }
 
     @Test
     void processUtilitiesExposeNativeCharsetJavaCommandAndSearchPath() {

--- a/tests/src/io.smallrye.common/smallrye-common-process/2.17.0/user-code-filter.json
+++ b/tests/src/io.smallrye.common/smallrye-common-process/2.17.0/user-code-filter.json
@@ -1,0 +1,10 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "io.smallrye.common.process.**"
+    }
+  ]
+}

--- a/tests/src/io.smallrye.common/smallrye-common-process/2.17.0/user-code-filter.json
+++ b/tests/src/io.smallrye.common/smallrye-common-process/2.17.0/user-code-filter.json
@@ -5,6 +5,9 @@
     },
     {
       "includeClasses" : "io.smallrye.common.process.**"
+    },
+    {
+      "includeClasses" : "io_smallrye_common.smallrye_common_process.**"
     }
   ]
 }

